cli/factory.rs

```diff
@@ -690,17 +690,19 @@ impl CliFactory {
       is_inspecting: self.options.is_inspecting(),
       is_npm_main: self.options.is_npm_main(),
       location: self.options.location_flag().clone(),
-      maybe_binary_npm_command_name: {
-        let mut maybe_binary_command_name = None;
-        if let DenoSubcommand::Run(flags) = self.options.sub_command() {
-          if let Ok(pkg_ref) = NpmPackageReqReference::from_str(&flags.script) {
-            // if the user ran a binary command, we'll need to set process.argv[0]
-            // to be the name of the binary command instead of deno
-            maybe_binary_command_name =
-              Some(npm_pkg_req_ref_to_binary_command(&pkg_ref));
-          }
-        }
-        maybe_binary_command_name
+      // if the user ran a binary command, we'll need to set process.argv[0]
+      // to be the name of the binary command instead of deno
+      maybe_binary_npm_command_name: match self.options.sub_command() {
+        DenoSubcommand::Run(flags) => flags
+          .npm_bin_command_name
```

Comment:
If we go with an env var we could capture it in `cli/args/mod.rs` and expose in `CliOptions`?

Reply:
Ah, good idea.